### ISSUE TITLE
Query regarding penalizing incorrect detections

### ISSUE THREAD

**Issue by prantiksen4** [experiencor/basic-yolo-keras#166]
My data set has only one object per image. The background of each image is white. Whereas my test set can have images on random background (includes text besides other shapes). 
My query is do you penalize incorrect detection on the validation set?
 I mean in case, my validation set had random objects and text and if they get wrongly detected as one of the classes, whether there would be any penalty imposed

**Comment by experiencor** [comment]
@prantiksen4 Yes, they do get penalized. You may consider adding random backgrounds to the images in the training set.

**Comment by prantiksen4** [comment]
Thanks for the reply. Actually a few undesired objects are also getting detected in my test set. My training set has nicely cropped images of just the object. So i suppose I need to have a few images similar to the test images so that it penalizes incorrect detection and learns weights that neglect them.
